### DISCUSSION TITLE
How challenging would it be to implement bidirectional syncing of notes to Onedrive using something like rclone?

### DISCUSSION THREAD

**Discussion by antomicblitz** [flxzt/rnote#691] [Ideas]
First off, thanks for making this project. I really admire the talent. I am curious if syncing to onenote or Google drive would be difficult to implement? I had high hopes that the developer of Xournal++ would make a user friendly implementation for that, but it seems he doesn't care to do it which is really a shame. I think having unified cloud syncing is an integral part to getting apps like this to be widely accepted as viable replacements to proprietary software.

**Comment by antomicblitz** [comment]
[https://rclone.org/bisync/]()
I was thinking that maybe integrating an existing tool like rclone bisync might be a promising approach? I don't know if that is the best approach.

**Comment by flxzt** [comment]
Yes, I am interested in implementing some sort of syncing.
I believe there are two topics here:

- the first is to make the app work together with software from cloud providers, like Google Drive or OneDrive, on a per-file level. I believe currently there are some issues with that, #461 is the tracking issue for that. "rclone bisync" looks interesting, as long as this is cloud-provider agnostic it might be worth considering. I would need to look into it more closely, what currently breaks cloud-syncing on the file-level.

- the second would be: enabling real-time syncing between different instances, like a collaborative editor. For that I already looked at some CRDT and peer-2-peer libraries that could enable syncing between different machine directly, without the need for a server. It should be doable, it needs quite a bit of work to make it possible. (also see issue #2,the progress for it will be tracked there).

**Reply by antomicblitz** [reply]
Here is the page. 
https://github.com/rclone/rclone

 They are compatible with almost every cloud syncing provider. I have tried rclone myself to mount my onedrive on my Linux desktop and sync between them, and it actually works! In basic scenarios, it works well, but I think in more complicated situations, like if the file is syncing at the same time it is being written, or if the file is being updated from multiple sources, it might crap out. But the tool exists so that is promising 🙂. I just wish it had a good GUI. But anyway I'm so glad to hear it is something you are thinking about. Those features would be really awesome! Thank you again for making this project. It is so exciting because it's really a 'David vs Goliath' project. We need this badly to loosen Microsofts despotic grip on note taking and office apps.

**Reply by antomicblitz** [reply]
Oh and this might also be an interesting read!

https://www.reddit.com/r/rclone/comments/rf2xjo/is_bisync_reliable_enough_in_rclone_beta/?utm_source=share&utm_medium=android_app&utm_name=androidcss&utm_term=1&utm_content=share_button
